### PR TITLE
Return non-zero value if part of segments fail to do incremental reco…

### PR DESCRIPTION
**Pull request by paul-guo-**
…very in gprecoverseg.

This helps script handling by checking return values.

### CODE REVIEW

**Comment by asimrp** [comment]
This change may affect behave tests, have you tried running them?  I'm not sure if PR pipeline runs behave tests.

The current PR pipeline failure is not related to this patch at all.

File: gpMgmt/bin/gppylib/operations/buildMirrorSegments.py [hunk 1]
```diff
@@ -329,6 +330,9 @@ def buildMirrors(self, actionName, gpEnv, gpArray):
             # Re-enable Ctrl-C
             signal.signal(signal.SIGINT, signal.default_int_handler)
 
+        if len(rewindFailedSegments) != 0:
```
[the remark this answers is not in the document]
Comment (paul-guo-) [reply gpMgmt/bin/gppylib/operations/buildMirrorSegments.py]: Here is the output of master without this patch. We can see that which segments failing to run pg_rewind are shown on the output as warning. With this patch, the warning messages are still shown but the return value is non-zero now.

```
20191226:13:36:42:012376 gprecoverseg:dev7:gpadmin-[INFO]:-Updating mirrors
20191226:13:36:42:012376 gprecoverseg:dev7:gpadmin-[INFO]:-Running pg_rewind on required mirrors
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[WARNING]:-Incremental recovery failed for dbid 5. You must use gprecoverseg -F to recover the segment.
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-Starting mirrors
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-era is 068c50aab6136cea_191226132221
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-Triggering FTS probe
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-******************************************************************
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-Updating segments for streaming is completed.
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-For segments updated successfully, streaming will continue in the background.
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-Use  gpstate -s  to check the streaming progress.
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-******************************************************************
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-==============================END ANOTHER RECOVER==========================================
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-******************************************************************
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-The rebalance operation has completed successfully.
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-There is a resynchronization running in the background to bring all
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-segments in sync.
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-Use gpstate -e to check the resynchronization progress.
20191226:13:36:43:012376 gprecoverseg:dev7:gpadmin-[INFO]:-******************************************************************
```

[the remark this answers is not in the document]
Comment (asimrp) [reply gpMgmt/bin/gppylib/operations/buildMirrorSegments.py]: That is a good improvement, thank you for the clarification.

**Comment by paul-guo-** [comment]
I've tested this using my dev pipeline. It's fine. So checking in the code into both master and 6. Thanks for reviewing.

**Pull request merged**